File: Python3/1461_check_if_a_string_contains_all_binary_codes_of_size_k.py
# https://leetcode.cn/problems/check-if-a-string-contains-all-binary-codes-of-size-k/

class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        record = set()
        for i in range(len(s)-k+1):
            record.add(s[i:i+k])
        return 2**k == len(record)

File: Cpp/1461_check_if_a_string_contains_all_binary_codes_of_size_k.cpp
// https://leetcode.cn/problems/check-if-a-string-contains-all-binary-codes-of-size-k/

#include <iostream>
#include <string>
#include <unordered_set>
#include <cmath>

using namespace std;

class Solution {
public:
    bool hasAllCodes(string s, int k) {
        unordered_set<string> record;
        for (int i = 0; i < (int)s.size()-k+1; ++i) {
            record.insert(s.substr(i, k));
        }
        return pow(2, k) == record.size();
    }
};

int main() {
    Solution sol;
    string s = "00110110";
    int k = 2;
    cout << sol.hasAllCodes(s, k) << '\n';
    return 0;
}